### PR TITLE
Jetpack Plans: Add HappyChat button to thank you page after Jetpack plan was acquired

### PR DESCRIPTION
**Pull request by oskosk**
* Adds a HappyChat Buttonin to the Jetpack ThankYou Card before the "Visit your site" button.
* Introduces an event `calypso_plans_autoconfig_chat_initiated' that' recorded when the button is clicked.

#### Testing instructions

1. Clone this branch and test it locally.
1. Log in into HappyChat staging as an operator (this is just to ensure there's at least one operator and the HappyChatButton shows. Otherwise you won't see it).
1. Connect a Jetpack site and acquire a paid plan.
1. Wait until you see the Thank you card, and until the "Ask a question" button is shown.

#### Screenshot

image

### CODE REVIEW

**Comment by chaselivingston** [comment]
Awesome to see this working! A couple of design suggestions, with the caveat that IANAD:

1. Move the prompt just below the feature list.
2. Make it a button that says "Have Questions?"

**Comment by oskosk** [comment]
Thanks for the feedback @chaselivingston. The PR is updated and it looks like this now.

image

**Comment by chaselivingston** [comment]
Could we make the button look similar to the "Visit Your Site" button perhaps?

**Comment by oskosk** [comment]
Indeed. looking like this now:

image

Would that be fine or we shoudl make the colors also be the same ?

**Comment by rickybanister** [comment]
@oskosk where's the design review?

Let's put this button inside the thank you card next to the visit site button. 

So `[Ask a question] [Visit your site]`

**Comment by chaselivingston** [comment]
> Let's put this button inside the thank you card next to the visit site button.

This is why I'm not a designer, didn't even think of that :).

**Comment by oskosk** [comment]
> @oskosk where's the design review?

@rickybanister The PR has been labeled as in need of design review even before it's ready for code review :(

I'll update it as you propose

File: client/my-sites/upgrades/checkout-thank-you/jetpack-thank-you-card.jsx [hunk 1]
```diff
@@ -527,14 +547,25 @@ class JetpackThankYouCard extends Component {
...
 
 		if ( 100 === progress ) {
-			return null;
+			return (
+				<div>
+					{ this.renderLiveChatButton() }
+					<a
+						className={ classNames( 'thank-you-card__button', { 'is-placeholder': ! buttonUrl } ) }
+						href={ buttonUrl }>
+						{ translate( 'Visit your site' ) }
```
Comment (a8ci18n) [inline client/my-sites/upgrades/checkout-thank-you/jetpack-thank-you-card.jsx]: Hi! I've found a possible matching string that has already been translated **36** times:
`translate( 'Visit Your Site' )` *ES Score: 11*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/my-sites/upgrades/checkout-thank-you/jetpack-thank-you-card.jsx [hunk 2]
```diff
@@ -307,6 +308,23 @@ class JetpackThankYouCard extends Component {
...
+
+	renderLiveChatButton() {
+		const isJetpackPaidPlan = true;
+		return isJetpackPaidPlan && (
+			<HappyChatButton
+				borderless={ false }
+				className="checkout-thank-you__happychat-button thank-you-card__button"
+				onClick={ this.onHappyChatButtonClick }
+			>
+				{ this.props.translate( 'Ask A Question' ) }
```
Comment (a8ci18n) [inline client/my-sites/upgrades/checkout-thank-you/jetpack-thank-you-card.jsx]: Hi! I've found a possible matching string that has already been translated **26** times:
`translate( 'Ask a question' )` *ES Score: 17*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/happychat-to-post-plans-screen)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by oskosk** [comment]
@rickybanister @evilluendas  I've updated the changes here to show the button inside the ThankYouCard.

Question, should the buttons be positioned side by side? Take in account that the happy chat button may be shown in an async manner due to a few reasons (lack of available operator by the time the card is first shown for example). Here's a gif showing the state transitions.

happychat

**Comment by oskosk** [comment]
About the button text, the i18n bot is suggesting "Visit Your Site" and "Ask a question". Different cases there. Is that OK ?

**Comment by jamilabreu** [comment]
@oskosk Just some quick input based on what I was imagining it would look like:

Heading -> Thanks for your purchase!
Subheading -> Interested in a personalized tour from one of our Happiness Engineers on how to best take advantage of Jetpack on [URL]?

Button -> Sure! I’d like to chat right now. (Action: Initiates HappyChat)
Button -> Sure! I’d like to schedule a chat. (Action: Initiates Calend.ly Appointment)
Button -> No thanks! Take me to my site

**Comment by oskosk** [comment]
@nosolosw would you mind giving this a look?

@rickybanister do you think this is going the right way ? I left a few questions for you about some details in a previous comment.

**Comment by nosolosw** [comment]
I'm heading to AFK for the next few weeks in a few minutes, but wanted to say that general code direction looks good.

Said that, you'll probably want a reviewer that can provide help during the next weeks :) I'd recommend keeping any of the following gentlemen @southp @mattwondra @unDemian @omarjackman in the loop.

**Comment by dllh** [comment]
I've asked @evilluendas to do a quick final pass on the design questions in the morning if @rickybanister doesn't beat him to it, and I've asked @unDemian to own this and help push it through to launch (hopefully in the next day or two) since @oskosk is afk for now.

**Comment by unDemian** [comment]
Sorry for the late response, code looking good and it works as expected.

**Pull request merged**